dali/test/python/decoder/test_image.py

```diff
@@ -65,31 +65,28 @@ def decoder_pipe(data_path, device, use_fast_idct=False, memory_stats=False,
 test_missnamed_path = {'jpeg', 'png', 'tiff', 'pnm', 'bmp'}
 
 
-def run_decode(data_path, batch, device, threads, memory_stats=False):
+def run_decode(_img_type, data_path, batch, device, threads, memory_stats=False):
     pipe = decoder_pipe(data_path=data_path, batch_size=batch, num_threads=threads, device_id=0,
                         device=device, memory_stats=memory_stats, prefetch_queue_depth=1)
     pipe.build()
     iters = math.ceil(pipe.epoch_size("Reader") / batch)
     for _ in range(iters):
-        pipe.run()
+        outs = pipe.run()
+        del outs
+    del pipe
 
 
 def test_image_decoder():
-    def log(img_type, size, device, threads):
-        pass
-
     for device in {'cpu', 'mixed'}:
         for batch_size in {1, 10}:
             for img_type in test_good_path:
                 for threads in {1, random.choice([2, 3, 4])}:
                     data_path = os.path.join(test_data_root, good_path, img_type)
-                    run_decode(data_path, batch_size, device, threads)
-                    yield log, img_type, batch_size, device, threads
+                    yield run_decode, img_type, data_path, batch_size, device, threads
```

Comment:
You need to adjust other invocations of `run_decode` like L242.

Reply:
Done.